### ISSUE TITLE
GOOD LUCK

### ISSUE THREAD

**Issue by xiangxinhello** [showlab/Tune-A-Video#98]

**Comment by xiangxin199621** [comment]
![Uploading sample-500.gif…]()

**Comment by xiangxin199621** [comment]
![sample-500](https://github.com/showlab/Tune-A-Video/assets/169245314/c7e68636-923a-4512-95ae-12b8b30d26e3)

**Comment by xiangxin199621** [comment]
![A jeep car is moving on the beach](https://github.com/showlab/Tune-A-Video/assets/169245314/f8c9d959-bcef-4f2c-a6b7-c70f436dce15)

**Comment by xiangxin199621** [comment]
![A Porsche car is moving on the beach](https://github.com/showlab/Tune-A-Video/assets/169245314/986c9f7e-1bf3-459d-9da1-d83d83ca88e2)

**Comment by xiangxin199621** [comment]
from tuneavideo.pipelines.pipeline_tuneavideo import TuneAVideoPipeline
from tuneavideo.models.unet import UNet3DConditionModel
from tuneavideo.util import save_videos_grid
import torch

pretrained_model_path = "/home/下载/Tune-A-Video-main/checkpoints/stable-diffusion-v1-4"
my_model_path = "/home/下载/Tune-A-Video-main/outputs/car-turn"
unet = UNet3DConditionModel.from_pretrained(my_model_path, subfolder='unet', torch_dtype=torch.float16).to('cuda')
pipe = TuneAVideoPipeline.from_pretrained(my_model_path, unet=unet, torch_dtype=torch.float16).to("cuda")
pipe.enable_xformers_memory_efficient_attention()
pipe.enable_vae_slicing()

prompt = "A Porsche car is moving on the beach"
ddim_inv_latent = torch.load(f"{my_model_path}/inv_latents/ddim_latent-500.pt").to(torch.float16)
video = pipe(prompt, latents=ddim_inv_latent, video_length=24, height=512, width=512, num_inference_steps=50, guidance_scale=12.5).videos

save_videos_grid(video, f"./{prompt}.gif")

**Comment by xiangxin199621** [comment]
> ![sample-500](https://private-user-images.githubusercontent.com/169245314/328815369-6dadf215-9d6c-4399-b488-7f62c2c2236c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxNjI3NjAsIm5iZiI6MTcxNTE2MjQ2MCwicGF0aCI6Ii8xNjkyNDUzMTQvMzI4ODE1MzY5LTZkYWRmMjE1LTlkNmMtNDM5OS1iNDg4LTdmNjJjMmMyMjM2Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOFQxMDAxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hM2UwZDE2Nzc2ZDY2NDhmYjI4YjhjMDg5NjNlN2VhZDkwNGI5NmU3NGE1NDI2MDhlMDE2Yzg2YjE4MmQwYmFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5Lt68PmV7BWkEgUC22jT76qHjBQcRzjUIkQaks3YeX8) [ ![sample-500](https://private-user-images.githubusercontent.com/169245314/328815369-6dadf215-9d6c-4399-b488-7f62c2c2236c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxNjI3NjAsIm5iZiI6MTcxNTE2MjQ2MCwicGF0aCI6Ii8xNjkyNDUzMTQvMzI4ODE1MzY5LTZkYWRmMjE1LTlkNmMtNDM5OS1iNDg4LTdmNjJjMmMyMjM2Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOFQxMDAxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hM2UwZDE2Nzc2ZDY2NDhmYjI4YjhjMDg5NjNlN2VhZDkwNGI5NmU3NGE1NDI2MDhlMDE2Yzg2YjE4MmQwYmFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5Lt68PmV7BWkEgUC22jT76qHjBQcRzjUIkQaks3YeX8) ](https://private-user-images.githubusercontent.com/169245314/328815369-6dadf215-9d6c-4399-b488-7f62c2c2236c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxNjI3NjAsIm5iZiI6MTcxNTE2MjQ2MCwicGF0aCI6Ii8xNjkyNDUzMTQvMzI4ODE1MzY5LTZkYWRmMjE1LTlkNmMtNDM5OS1iNDg4LTdmNjJjMmMyMjM2Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOFQxMDAxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hM2UwZDE2Nzc2ZDY2NDhmYjI4YjhjMDg5NjNlN2VhZDkwNGI5NmU3NGE1NDI2MDhlMDE2Yzg2YjE4MmQwYmFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5Lt68PmV7BWkEgUC22jT76qHjBQcRzjUIkQaks3YeX8) [ ](https://private-user-images.githubusercontent.com/169245314/328815369-6dadf215-9d6c-4399-b488-7f62c2c2236c.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxNjI3NjAsIm5iZiI6MTcxNTE2MjQ2MCwicGF0aCI6Ii8xNjkyNDUzMTQvMzI4ODE1MzY5LTZkYWRmMjE1LTlkNmMtNDM5OS1iNDg4LTdmNjJjMmMyMjM2Yy5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOFQxMDAxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hM2UwZDE2Nzc2ZDY2NDhmYjI4YjhjMDg5NjNlN2VhZDkwNGI5NmU3NGE1NDI2MDhlMDE2Yzg2YjE4MmQwYmFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5Lt68PmV7BWkEgUC22jT76qHjBQcRzjUIkQaks3YeX8)
> 
> pretrained_model_path: "./checkpoints/stable-diffusion-v1-4" output_dir: "./outputs/man-surfing"
> 
> train_data: video_path: "data/man-surfing.mp4" prompt: "a man is surfing" n_sample_frames: 24 width: 512 height: 512 sample_start_idx: 0 sample_frame_rate: 1
> 
> validation_data: prompts: - "a panda is surfing" - "a boy, wearing a birthday hat, is surfing" - "a raccoon is surfing, cartoon style" - "Iron Man is surfing in the desert" video_length: 24 width: 512 height: 512 num_inference_steps: 50 guidance_scale: 12.5 use_inv_latent: True num_inv_steps: 50
> 
> learning_rate: 3e-5 train_batch_size: 1 max_train_steps: 500 checkpointing_steps: 1000 validation_steps: 100 trainable_modules:
> 
> * "attn1.to_q"
> * "attn2.to_q"
> * "attn_temp"
> 
> seed: 33 mixed_precision: fp16 use_8bit_adam: False gradient_checkpointing: True enable_xformers_memory_efficient_attention: True
> 
> tokenizers 0.13.3 pypi_0 pypi torch 1.13.1+cu116 pypi_0 pypi torchaudio 0.13.1+cu116 pypi_0 pypi torchvision 0.14.1+cu116 pypi_0 pypi transformers 4.25.1 pypi_0 pypi

How to fine tune?
